In [2]:
import pandas as pd

In [68]:
crime = pd.read_csv('crime.csv')

total = len(crime)

#Sorted Alphabetically
#crime = crime.sort_values('INCIDENTNEIGHBORHOOD')['INCIDENTNEIGHBORHOOD'].value_counts(sort = False).rename_axis("Neighborhood")
#print(crime.to_string())

#Sorted in decending order
crime = crime['INCIDENTNEIGHBORHOOD'].value_counts().rename_axis("Neighborhood")
print(crime.to_string())

Neighborhood
Central Business District    210
South Side Flats             122
Carrick                      118
East Liberty                  91
Homewood North                83
Brookline                     76
Brighton Heights              73
Mount Washington              72
East Hills                    67
Beechview                     66
Sheraden                      64
Knoxville                     62
East Allegheny                56
Lincoln-Lemington-Belmar      55
Shadyside                     54
Marshall-Shadeland            54
Bloomfield                    52
Bluff                         51
Hazelwood                     51
Squirrel Hill South           51
Homewood South                50
Crawford-Roberts              48
Perry South                   48
Crafton Heights               45
Allentown                     44
Larimer                       42
Middle Hill                   40
South Side Slopes             39
Squirrel Hill North           39
Garfield                      

# The 3 Neighborhoods with the most crime:

- Central Business District:    210 Cases
- South Side Flats:             122 Cases
- Carrick:                      118 Cases

# The 7 Neighbourhoods with the least crime:

- Esplen:                         3 Cases
- Glen Hazel:                     2 Cases
- Regent Square:                  2 Cases
- Mount Oliver Borough:           2 Cases
- Outside County:                 2 Cases
- Summer Hill:                    2 Cases
- Ridgemont:                      1 Cases

In [70]:
#crime rate for top 3 neighborhoods compared to all crime in 30-day period
CBDcr = 210/total *100 
SSFcr = 122/total *100
CARcr = 118/total *100

print ("Central Business District Crime Rate:", CBDcr, "%")
print ("South Side Flats Crime Rate:", SSFcr, "%")
print ("Carrick Crime Rate:", CARcr, "%")

Central Business District Crime Rate: 7.211538461538461 %
South Side Flats Crime Rate: 4.189560439560439 %
Carrick Crime Rate: 4.052197802197802 %


In [71]:
#crime rate for bottom 7 neighborhoods compared to all crime in 30-day period
ESPcr = 3/total *100
GHcr = 2/total *100
RIDcr = 1/total *100

print ("Esplen Crime Rate:", ESPcr, "%")
print ("Glen Hazel, Regent Square, Mt. Oliver Boro, Outside County, and Summer Hill Crime Rates:", GHcr, "%")
print ("Ridgemont Crime Rate:", RIDcr, "%")

Esplen Crime Rate: 0.10302197802197803 %
Glen Hazel, Regent Square, Mt. Oliver Boro, Outside County, and Summer Hill Crime Rates: 0.06868131868131869 %
Ridgemont Rate: 0.034340659340659344 %


# Looking at Population data compared to Crime data:

In [76]:
#Population dataframe
population = pd.read_csv('total-population.csv')

#Population of top 3 neighborhoods (crime rate)
print(population.loc[population['Neighborhood'] == "Central Business District"])
print ("")
print(population.loc[population['Neighborhood'] == "South Side Flats"])
print("")
print(population.loc[population['Neighborhood'] == "Carrick"])

                 Neighborhood  Id  Estimate; Total  Margin of Error; Total
16  Central Business District  17           4060.0                458.0262

        Neighborhood  Id  Estimate; Total  Margin of Error; Total
72  South Side Flats  73           6292.0              747.592135

   Neighborhood  Id  Estimate; Total  Margin of Error; Total
15      Carrick  16          10122.0              826.666196


In [78]:
#Population of top 7 neighborhoods (crime rate)

print(population.loc[population['Neighborhood'] == "Esplen"])
print ("")
print(population.loc[population['Neighborhood'] == "Glen Hazel"])
print("")
print(population.loc[population['Neighborhood'] == "Regent Square"])
print ("")
print(population.loc[population['Neighborhood'] == "Mount Oliver Borough"])
print ("")
print(population.loc[population['Neighborhood'] == "Outside County"])
print("")
print(population.loc[population['Neighborhood'] == "Summer Hill"])
print("")
print(population.loc[population['Neighborhood'] == "Ridgemont"])

   Neighborhood  Id  Estimate; Total  Margin of Error; Total
30       Esplen  31            309.0                   118.0

   Neighborhood  Id  Estimate; Total  Margin of Error; Total
35   Glen Hazel  36            716.0                   231.0

     Neighborhood  Id  Estimate; Total  Margin of Error; Total
66  Regent Square  67           1033.0                   101.0

            Neighborhood  Id  Estimate; Total  Margin of Error; Total
52  Mount Oliver Borough  53           3399.0              462.961121

Empty DataFrame
Columns: [Neighborhood, Id, Estimate; Total, Margin of Error; Total]
Index: []

   Neighborhood  Id  Estimate; Total  Margin of Error; Total
81  Summer Hill  82           1191.0                    92.0

   Neighborhood  Id  Estimate; Total  Margin of Error; Total
67    Ridgemont  68            314.0                   109.0


# The Population of the 3 Neighborhoods with the most crime:

- Central Business District:    4060 People
- South Side Flats:             6292 People
- Carrick:                      10122 People

# The Population of the 7 Neighbourhoods with the least crime:

- Esplen:                         309 People
- Glen Hazel:                     716 People
- Regent Square:                  1033 People
- Mount Oliver Borough:           3399 People
- Outside County:                 No Data
- Summer Hill:                    1191 People
- Ridgemont:                      314 People